packages/arb-token-bridge-ui/src/components/RecoverFunds.tsx

```diff
@@ -0,0 +1,530 @@
+      gasEstimation.gasLimit, // gasLimit
+      gasEstimation.maxFeePerGas, // maxFeePerGas
+      0, // tokenTotalFeeAmount
+      '0x', // data
+      {
+        from: signerAddress
+      }
+    )
+  }
+
+  const inbox = Inbox__factory.connect(inboxAddress, childProvider)
+  return inbox.connect(signer).unsafeCreateRetryableTicket(
+    destinationAddress, // to
+    l2CallValue, // l2CallValue
+    gasEstimation.maxSubmissionCost, // maxSubmissionCost
+    destinationAddress, // excessFeeRefundAddress
+    destinationAddress, // callValueRefundAddress
+    gasEstimation.gasLimit, // gasLimit
+    gasEstimation.maxFeePerGas, // maxFeePerGas
+    '0x', // data
+    {
+      from: signerAddress,
+      value: 0
+    }
+  )
+}
+
+async function prepareTransaction({
+  address,
+  destinationAddress,
+  childProvider,
+  parentProvider,
+  balanceToRecover
+}: {
+  address: string
+  destinationAddress: string
+  childProvider: StaticJsonRpcProvider
+  parentProvider: StaticJsonRpcProvider
+  balanceToRecover: BigNumber
+}) {
+  // We estimate gas usage
+  const parentToChildMessageGasEstimator = new ParentToChildMessageGasEstimator(
+    childProvider
+  )
+  const aliasedAddress = getAliasedAddress(address)
+
+  const gasEstimation = await parentToChildMessageGasEstimator.estimateAll(
+    {
+      from: aliasedAddress,
+      to: destinationAddress,
+      l2CallValue: balanceToRecover,
+      excessFeeRefundAddress: destinationAddress,
+      callValueRefundAddress: destinationAddress,
+      data: '0x'
+    },
+    await getBaseFee(parentProvider),
+    parentProvider
+  )
+
+  // And we send the request through the method unsafeCreateRetryableTicket of the Inbox contract
+  // We need this method because we don't want the contract to check that we are not sending the l2CallValue
+  // in the "value" of the transaction, because we want to use the amount that is already on child chain
+  const l2CallValue = balanceToRecover
+    .sub(gasEstimation.maxSubmissionCost)
+    .sub(gasEstimation.gasLimit.mul(gasEstimation.maxFeePerGas))
+
+  return {
+    l2CallValue,
+    gasEstimation
+  }
+}
+
+async function recoverFunds({
+  address,
+  destinationAddress,
+  childProvider,
+  parentProvider,
+  balanceToRecover,
+  inboxAddress,
+  signer
+}: {
+  address: string
+  destinationAddress: string
+  childProvider: StaticJsonRpcProvider
+  parentProvider: StaticJsonRpcProvider
+  balanceToRecover: BigNumber
+  inboxAddress: string
+  signer: Signer
+}) {
+  const { gasEstimation, l2CallValue } = await prepareTransaction({
+    address,
+    destinationAddress,
+    childProvider,
+    parentProvider,
+    balanceToRecover
+  })
+
+  return createRetryableTicket({
+    childProvider,
+    destinationAddress,
+    gasEstimation,
+    inboxAddress,
+    l2CallValue,
+    signer,
+    signerAddress: address
+  })
+}
+
+export function getAliasedAddress(address: string) {
+  return new Address(address).applyAlias().value
+}
```

Comment:
We have also defined this function in `recover-funds.ts`.

Reply:
I have removed `recover-funds.ts`